cts.json

```diff
@@ -2115,6 +2115,160 @@
       "result": [
         {"d": "f"}
       ]
+    },
+    {
+      "name": "filter, length function, string data",
+      "selector" : "$[?length(@.a)>=2]",
+      "document" : [{"a": "ab"}, {"a": "d"}],
+      "result": [
+        {"a": "ab"}
+      ]
+    },
+    {
+      "name": "filter, length function, array data",
+      "selector" : "$[?length(@.a)>=2]",
+      "document" : [{"a": [1,2,3]}, {"a": [1]}],
+      "result": [
+        {"a": [1,2,3]}
+      ]
+    },
+    {
+      "name": "filter, length function, missing data",
+      "selector" : "$[?length(@.a)>=2]",
+      "document" : [{"d": "f"}],
+      "result": []
+    },
+    {
+      "name": "filter, length function, non-array/string arg",
+      "selector" : "$[?length(1)>=2]",
+      "invalid_selector": true
+    },
+    {
+      "name": "filter, length function, result must be compared",
+      "selector" : "$[?length(@.a)]",
+      "invalid_selector": true
+    },
+    {
+      "name": "filter, count function",
+      "selector" : "$[?count(@..*)>2]",
+      "document" : [
+        {"a": [1, 2, 3]},
+        {"a": [1], "d": "f"},
+        {"a": 1, "d": "f"}
+      ],
+      "result": [
+        {"a": [1,2,3]},
+        {"a": [1], "d":"f"}
+      ]
+    },
+    {
+      "name": "filter, count function, non-array/string arg",
+      "selector" : "$[?count(1)>2]",
+      "invalid_selector": true
+    },
+    {
+      "name": "filter, count function, result must be compared",
+      "selector" : "$[?count(@..*)]",
+      "invalid_selector": true
+    },
+    {
+      "name": "filter, match function, found match",
+      "selector" : "$[?match(@.a, '2022-01-[0-9]{2}')]",
```

Comment:
Although (possibly except for the single quotes, see below) this is a correct regexp, it's a bit peculiar because it allows dates such as `2022-01-99`. This would be a small distraction to anyone reading the tests. This is exacerbated by a number of tests using the same regexp. Why not avoid this by choosing a different regexp?

Reply:
It was familiar-enough.  Happy to change to something else.  Open to proposals.

Reply:
I would go for something trivial like `a.*` matching `ab` or some such.

Reply:
I'll try to rework these, then.